homeassistant/components/zha/cover.py

```diff
@@ -128,6 +129,12 @@ def async_set_position(self, attr_id, attr_name, value):
             self._state = STATE_CLOSED
         elif self._current_position == 100:
             self._state = STATE_OPEN
+        elif (
+            self._requested_position
+            and self._current_position == self._requested_position
+        ):
+            self._state = STATE_OPEN
```

Comment:
I believe that some covers are reporting discrete values. In these devices, when you require (ie) a set_position=70 the cover moves to position but it reports (ie) a current_position=69.
If you raise a little more, maybe you get a current_position=75, but you can get all the 0-100 values.

For these devices this approach would not fix the issue.

Reply:
@m67hoff  Ideally, we would like to have a state Partly_Open which would be anything between Close and Open, but that states doesn't exist. That's why, even opened at 2%, a blind is still Open and not Close.